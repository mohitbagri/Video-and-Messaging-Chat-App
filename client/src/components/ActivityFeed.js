import React from "react";
import { useState, useEffect } from "react";
import axios from "axios";
import { API_URL } from './Utils.js';

const ActivityFeed = () => {
    const loadStatus = async () => {
        console.log("do we get in the func")
        var username = localStorage.getItem("username");
        // /FETCHUSERCONTACTS 
        const contacts = await axios.post(API_URL + '/fetchUserContacts', {username})
        const data = "username=" + username + "&contacts=" + '...';
        const result = await axios.post(
            API_URL + '/activityFeed',
            data
        );
    };

    return (
        <div>
            <h1>Activity Feed View</h1>
            <h4>View Your Friends' Statuses</h4>
            <form>
                <button onClick={() => loadStatus()}>Click to load statuses</button>
            </form>
        </div>
    );
};

export default ActivityFeed;
