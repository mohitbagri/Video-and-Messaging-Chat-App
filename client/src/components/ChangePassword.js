import React, { Component } from "react";
import axios from 'axios';
import { API_URL } from './Utils.js';

class ChangePassword extends Component {

    constructor(props) {
        super(props);
        this.changePwd = this.changePwd.bind(this);
        this.state = {oldPassword: '',
                      newPassword: '',
                      verifiedNewPassword:'Unsuccesful',
                      verifiedSecurityQuestion:'Unsuccesful'
            };
    
        this.handleOldPasswordChange = this.handleOldPasswordChange.bind(this);
        this.handleNewPasswordChange = this.handleNewPasswordChange.bind(this);
        this.validateNewPassWord = this.validateNewPassWord.bind(this);
        this.validateSecurityQuestion = this.validateSecurityQuestion.bind(this);
      }
    
       async changePwd() {

        console.log("old pwd is " + this.state.oldPassword + " new pwd is" + this.state.newPassword);
        await this.validateNewPassWord();
        await this.validateSecurityQuestion();
        console.log(" I am in changePwd and " + this.state.verifiedNewPassword + "^^" + this.state.verifiedSecurityQuestion);
        if (this.state.verifiedNewPassword === 'Success' && this.state.verifiedSecurityQuestion == 'Success'){
        const data =  "username="+localStorage.getItem('username')+"&newPassword="+this.state.newPassword; 
        const result =  await axios.post(`${API_URL}/updatePassword`, data);

           var messageStatus = result.data.message;
           console.log(messageStatus);
           alert('changePwd!');

        }else{
            this.newPassword='';
        }
        
        

      }

      async validateSecurityQuestion(){
        var SecurityQuestion = document.getElementById("security").value;
        const data =  "username="+localStorage.getItem('username')+"&SecurityQuestion="+SecurityQuestion; 
        console.log("in validateSQ " + SecurityQuestion);

        const result1 =  await axios.post(`${API_URL}/validateSecurityQuestion`, data);
        console.log("Result is "+ result1.data.message);
        this.state.verifiedSecurityQuestion = result1.data.message;
        console.log(" this.state.verifiedSecurityQuestion" +  this.state.verifiedSecurityQuestion);

        if (this.state.verifiedSecurityQuestion !='Success'){

            alert("Wrong security question entered!");
        }


      }

      async validateNewPassWord(){
        var value= this.state.newPassword;
        console.log("we are in validate password "+ value);
        var re = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/;
        const isOk = re.test(value);
        var reason='';
        
        if(!isOk) {

            var checkLowerCaseRe= /(?=.*[a-z])/;

            var isLower= checkLowerCaseRe.test(value);
            if(!isLower){
                reason += "No Lower Case\n";
            }

            var checkUpperCaseRe= /(?=.*[A-Z])/;

            var isUpper= checkUpperCaseRe.test(value);
            if(!isUpper){
                reason += "No Upper Case\n";
            }

            var checkDigitseRe= /(?=.*\d)/ ;

            var isDigit= checkDigitseRe.test(value);
            if(!isDigit){
                reason += "No Digit\n";
            }

            if (value.length < 6 || value.length > 20 ){
                reason +=  "Length not between 6 and 20\n";
            }
            alert("New Password invalid\n"+ reason);
        }
        else{
            this.state.verifiedNewPassword = 'Success';
        }
      }  

      handleOldPasswordChange(event) {
        this.setState({
            oldPassword: event.target.value
        });
       
      }   
      
      handleNewPasswordChange(event) {
        this.setState({
            newPassword: event.target.value
        });
       
      }   

  render() {
    return (
      <div>
        <h2>change password page </h2>
        
    

     <div className="form-group">
          <label>Security Question </label>
          <input type="security" id="security" className="form-control" placeholder=" Enter Security Question" />
      </div>

     <div className="form-group">
          <label>New Password</label>
          <input type="password" value={this.state.newPassword} onChange={this.handleNewPasswordChange} className="form-control" placeholder="Password" />
     </div>
     
        <button onClick={this.changePwd}>
      Click me!
    </button>
        
      </div>
    );
  }
}
 
export default ChangePassword;
