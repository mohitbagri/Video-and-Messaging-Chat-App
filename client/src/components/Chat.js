import React, { Component } from "react";
import ReactPlayer from "react-player";
import ReactDOM from 'react-dom';
import { API_URL } from './Utils.js';


 
class Chat extends React.Component {
    constructor(props) {
      super(props);
      this.state = {file: '',imagePreviewUrl: ''};
    }
  
   
    _handleSubmit(e) {
        console.log('handle uploading-', this.state.file);
      e.preventDefault();      
    }
    
  
    _handleImageChange(e) {
        console.log("im in handle image chaneg");
        console.log("my location is " +window.location.href);
      e.preventDefault();
  
      let reader = new FileReader();
      let file = e.target.files[0];
  
      reader.onloadend = () => {
        this.setState({
          file: file,
          imagePreviewUrl: reader.result
        });
      }
  
      reader.readAsDataURL(file)
    }
  
    render() {
      let {imagePreviewUrl} = this.state;
      let $imagePreview = null;
      if (imagePreviewUrl) {
          console.log("imagepreview url is " + imagePreviewUrl.substring(0,25));
          var fileType = imagePreviewUrl.substring(0,25);
          fileType = fileType.split("/")[0];
          console.log("filetype is " + fileType);
          fileType = fileType.split(":")[1];
          console.log("final filetype is  "+ fileType);

          
          if (fileType == "video"){
            $imagePreview = <video width="750" height="500" controls >
            <source src= {imagePreviewUrl} type="video/mp4"/>
            </video>
          }
          if(fileType == "image"){
            $imagePreview = (<img src={imagePreviewUrl} width="100px" height="100px" />);
          }
        

        if (fileType =="audio"){
           $imagePreview = <audio src={imagePreviewUrl} controls />
        }
        

      } else {
        $imagePreview = (<div className="previewText">Please select the file for Preview</div>);
      }
  
      return (
        <div className="previewComponent">
          <form onSubmit={(e)=>this._handleSubmit(e)}>
            <input className="fileInput" 
              type="file" 
              onChange={(e)=>this._handleImageChange(e)} />
            <button className="submitButton" 
              type="submit" 
              onClick={(e)=>this._handleSubmit(e)}>Upload Multimedia file</button>
          </form>
          <div className="imgPreview">
            {$imagePreview}
          </div>
        </div>
      )
    }
  }
 
export default Chat;
