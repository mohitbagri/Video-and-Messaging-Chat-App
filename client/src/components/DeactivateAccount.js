import React, { Component } from "react";
import axios from 'axios';
import { API_URL } from './Utils.js';

class DeactivateAccount extends Component {
    constructor(props) {
        super(props);
        this.deactivate = this.deactivate.bind(this);
      }
    
      async deactivate() {
          console.log("i am in deactive acc" + localStorage.getItem('username'));
          const data =  "username="+localStorage.getItem('username'); 
          const result =  await axios.post(API_URL + '/changeAccountStatus', data);
          var messageStatus = result.data.message;
          console.log(messageStatus);
          if (messageStatus ==='Success'){
            alert('Changed account status successfully!');
          }else{
              alert("something went wrong!");
          }
        
      }
      
  render() {
    return (
      <div>
        <h2>activate/deactivate account</h2>
       
        <button onClick={this.deactivate}>   
      Click me!
    </button>
      </div>
    );
  }
}
 
export default DeactivateAccount;
