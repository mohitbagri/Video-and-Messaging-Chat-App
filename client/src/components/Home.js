import React, { Component } from "react";
import fb_default from "./fb_default.jpg";
import axios from 'axios';
import { API_URL } from './Utils.js';

class Home extends Component {

  render() {
      console.log("home page " + localStorage.getItem('username'));
    return (
      <div>
        <h2>HELLO</h2>
        <img src={fb_default} height='200px' width='200px' alt="fireSpot"/>
 
        
      </div>
    );
  }
}
 
export default Home;