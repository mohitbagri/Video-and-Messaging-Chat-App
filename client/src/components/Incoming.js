// Credit: Phil Nash (Twilio)

import React from "react";

const Incoming = ({ connection, device, socketRef, caller, setSocketCallUpdates }) => {
  const acceptConnection = () => {
    connection.accept();
    setSocketCallUpdates('')
  };
  const rejectConnection = () => {
    socketRef.current.emit('rejecting', {rejectedEntity: `${caller}`, rejectingEntity:`${localStorage.getItem('username')}`, room: `${caller}_updates` })
    setSocketCallUpdates('')
    connection.reject();
  };
  return (
    <>
      <button onClick={acceptConnection}>Accept</button>
      <button onClick={rejectConnection}>Reject</button>
    </>
  );
};

export default Incoming;