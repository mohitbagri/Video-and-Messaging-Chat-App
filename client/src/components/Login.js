import React, { Component, Fragment } from 'react';
import { useHistory } from 'react-router-dom';
import { withRouter } from "react-router-dom";
import LockScreen from "react-lock-screen";
import axios from 'axios';
import { Button, Navbar } from 'react-bootstrap';
import { API_URL } from './Utils.js';

const getLockScreenUi = setLock => {
      return (
        <div className="react-lock-screen__ui" style={{width: "800px",  margin: "0 auto"}}>
            
          <img
            width="32"
            src="https://cdn3.iconfinder.com/data/icons/wpzoom-developer-icon-set/500/102-256.png"
            alt="lock"
          />
          <p>Just to be safe, we locked the screen</p>
          <button onClick={() => setLock(false)}>unlock</button>
        </div>
      );
    };

class Login extends Component{

    constructor(props) {
        super(props);
    
        this.state = {
            value: '',
            failedLoginAttempts: 0,
           
    
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSignInClick = this.handleSignInClick.bind(this);
    }

        

    
   handleChange(event) {
    this.setState({
        value: event.target.value
        });
    }    

    async handleSignInClick(event) {

        document.getElementById("pwd").value = "";
        /*

        To check a password between 6 to 20 characters which contain at least 
        one numeric digit, one uppercase and one lowercase letter
        */

        event.preventDefault();
    
        const { value } = this.state;
        //const re = new RegExp("^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,32}$");
        //var re = /^[A-Z]*$/;
        var re = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/;
        const isOk = re.test(value);
        var reason='';
    
    
        if(!isOk) {
            var checkLowerCaseRe= /(?=.*[a-z])/;

            var isLower= checkLowerCaseRe.test(value);
            if(!isLower){
                reason += "No Lower Case\n";
                console.log("not lower");
            }

            var checkUpperCaseRe= /(?=.*[A-Z])/;

            var isUpper= checkUpperCaseRe.test(value);
            if(!isUpper){
                reason += "No Upper Case\n";
            }

            var checkDigitseRe= /(?=.*\d)/ ;

            var isDigit= checkDigitseRe.test(value);
            if(!isDigit){
                reason += "No Digit\n";
            }

            if (value.length < 6 || value.length > 20 ){
                reason +=  "Length not between 6 and 20\n";
            }


            console.log("this is reason after everything" + reason);

            this.state.failedLoginAttempts = this.state.failedLoginAttempts + 1;

            var outputReason = 'Password is weak\n' + reason + 'Number of Attempts Used: ' + this.state.failedLoginAttempts;
            
            return alert(outputReason);
            
        }
    
        if(isOk){
            
            alert('A password was submitted that was ' + value.length + ' characters long.');
            var username= document.getElementById("username").value;
            var password = value;
            const data = "username="+username+"&password="+password; 

            const result =  await axios.post(API_URL + '/loginUser', data);

        
            var messageStatus = result.data.message;

            console.log(" THIS IS MESSAGE STATUS " + messageStatus);
            
            if (messageStatus=='Success'){
                alert('Welcome back '+ username);
                localStorage.setItem('username', username);                
                this.props.history.push({
                    pathname: '/MainView',
                    customNameData: username,
                  });
                  
            }else if (messageStatus == 'Locked out'){
                alert('Your account is locked. Please try again after 30 minutes');
            }else{
                alert("Sorry buddy you are not a registered user/entered wrong password");
            }
                
        }
        
    }

    handleSignUplick = () => {
        this.props.history.push("/UserSignUp");
    }


  render() {
      
    return (
        <LockScreen timeout={500000} ui={getLockScreenUi}>
            
      <form>
      <Navbar location={this.props.location}/>
      <h3>Chat App!</h3>
      <h4>Login / Register</h4>
      <div className="form-group">
      <div className="form-group">
          <label>Username </label>
          <input type="username" id="username" className="form-control" placeholder=" Enter Username" />
      </div>
      </div>

      <div className="form-group">
          <label>Password</label>
          <input type="password" id="pwd" value={this.state.value} onChange={this.handleChange} className="form-control" placeholder="Password" />
        </div>
        <button onClick={this.handleSignInClick}>
      Sign in!
    </button>
        <button onClick={this.handleSignUplick}>
      Sign up!
    </button>
        
     
    </form>
    </LockScreen>
    
  
     
      );
      };
}

export default withRouter(Login);