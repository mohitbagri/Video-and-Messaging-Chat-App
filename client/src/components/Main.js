import React, { Component } from "react";
import { useLocation } from "react-router-dom";
import { API_URL } from './Utils.js';

import {
  Route,
  NavLink,
  HashRouter
} from "react-router-dom";
import Home from "./Home";
import { withRouter } from "react-router-dom";
import ChangePassword from "./ChangePassword";
import DeactivateAccount from "./DeactivateAccount";
import MainView from "./MainView";
import axios from 'axios';
import fb_default from "./fb_default.jpg";


 
class Main extends Component {

    constructor(props) {
        super(props);

        this.state = {
            profilePic: '',
            registrationTime: '',
            hasFetchedDetails: false
           
        };
        this.helper = this.helper.bind(this);
     
    }

    componentDidMount(){    
            this.helper();
    }
    
    async helper(event){
        
       const getProfile = await axios.get(API_URL + '/profileDetails', { params: { username: localStorage.getItem('username') } });
       
       var profileUserName =  getProfile.data.data.user;
       this.state.registrationTime = getProfile.data.data.registrationTime
       this.state.registrationTime = new Date(this.state.registrationTime).toLocaleDateString("en-US")
       this.state.hasFetchedDetails = true;
       this.state.profilePic =  getProfile.data.data.profilePicture;
       this.setState({registrationTime: this.state.registrationTime});
    }

    handleMainView = () => {
        this.props.history.push("/MainView");
    }
    
  render() {    
    return (
        
        <HashRouter>
        <div>
          <h1>{localStorage.getItem('username')} Profile Page</h1>
         
        <button onClick={this.handleMainView}>
            Go back to main view 
        </button>
          { this.state.registrationTime && 
                <div>registration date {this.state.registrationTime} 
                        <img src={this.state.profilePic} height='100px' width='100px' alt="fireSpot"/>
                </div>
            }          
          <ul className="header">
            <li><NavLink to="/ChangePassword">Change Password</NavLink></li>
            <li><NavLink to="/DeactivateAccount">Deactivate Account</NavLink></li>
          </ul>
          <div className="content">
            <Route path="/ChangePassword" component={ChangePassword}/>
            <Route path="/DeactivateAccount" component={DeactivateAccount}/>
             
          </div>
          
        </div>
        </HashRouter>
    );
  }
}
 
export default withRouter(Main);