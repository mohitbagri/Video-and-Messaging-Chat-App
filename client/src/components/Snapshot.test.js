import React from "react";
import renderer from "react-test-renderer";
import ActivityFeed from "./ActivityFeed.js";
import ChangePassword from "./ChangePassword.js";
import Chat from "./Chat.js";
// import ChatRoom from "./ChatRoom.js";
import DeactivateAccount from "./DeactivateAccount.js";
import Home from "./Home.js";
import Incoming from "./Incoming.js";
import Login from "./Login.js";
import Main from "./Main.js";
import MainView from "./MainView.js";
import OnCall from "./OnCall.js";
import Phone from "./Phone.js";
import PostStatus from "./PostStatus.js";
// import Timer from "./Timer.js";
import states from "./twilioStates.js";
import UserSignUp from "./UserSignUp.js";
// import ViewStatus from "./ViewStatus.js";

describe("ActivityFeed.js test", () => {
    it("snapshot test", () => {
        const tree = renderer.create(<ActivityFeed />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("ChangePassword.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<ChangePassword />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Chat.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<Chat />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// describe("ChatRoom.js test", () => {
//     it("snapshot test", () => {
//         const tree = renderer
//             .create(<ChatRoom />)
//             .toJSON();
//         expect(tree).toMatchSnapshot();
//     });
// });

describe("DeactivateAccount.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<DeactivateAccount />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Home.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<Home />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Incoming.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<Incoming />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Login.js test", () => {
    it("snapshot test", () => {
        let wrapper = <Login.WrappedComponent />;
        expect(wrapper).toMatchSnapshot();
    });
});

describe("Main.js test", () => {
    it("snapshot test", () => {
        let wrapper = <Main.WrappedComponent />;
        expect(wrapper).toMatchSnapshot();
    });
});

describe("MainView.js test", () => {
    it("snapshot test", () => {
        let wrapper = <MainView.WrappedComponent />;
        expect(wrapper).toMatchSnapshot();
    });
});

describe("OnCall.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<OnCall />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("Phone.js test", () => {
    it("snapshot test", () => {
        const tree = renderer
            .create(<Phone />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("PostStatus.js test", () => {
    it("snapshot test", () => {
        let wrapper = <PostStatus.WrappedComponent />;
        expect(wrapper).toMatchSnapshot();
    });
});

// describe("Timer.js test", () => {
//     it("snapshot test", () => {
//         const tree = renderer
//             .create(<Timer />)
//             .toJSON();
//         expect(tree).toMatchSnapshot();
//     });
// });

describe("twilioStates.js test", () => {
    it("snapshot test", () => {
        let wrapper = <states />;
        expect(wrapper).toMatchSnapshot();
    });
});


describe("UserSignUp.js test", () => {
    it("snapshot test", () => {
        let wrapper = <UserSignUp.WrappedComponent />;
        expect(wrapper).toMatchSnapshot();
    });
});

// describe("ViewStatus.js test", () => {
//     it("snapshot test", () => {
//         const tree = renderer
//             .create(<ViewStatus />)
//             .toJSON();
//         expect(tree).toMatchSnapshot();
//     });
// });
