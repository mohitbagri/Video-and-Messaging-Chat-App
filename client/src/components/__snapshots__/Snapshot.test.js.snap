// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityFeed.js test snapshot test 1`] = `
<div>
  <h1>
    Activity Feed View
  </h1>
  <h4>
    View Your Friends' Statuses
  </h4>
  <form>
    <button
      onClick={[Function]}
    >
      Click to load statuses
    </button>
  </form>
</div>
`;

exports[`ChangePassword.js test snapshot test 1`] = `
<div>
  <h2>
    change password page 
  </h2>
  <div
    className="form-group"
  >
    <label>
      Security Question 
    </label>
    <input
      className="form-control"
      id="security"
      placeholder=" Enter Security Question"
      type="security"
    />
  </div>
  <div
    className="form-group"
  >
    <label>
      New Password
    </label>
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <button
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;

exports[`Chat.js test snapshot test 1`] = `
<div
  className="previewComponent"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="fileInput"
      onChange={[Function]}
      type="file"
    />
    <button
      className="submitButton"
      onClick={[Function]}
      type="submit"
    >
      Upload Multimedia file
    </button>
  </form>
  <div
    className="imgPreview"
  >
    <div
      className="previewText"
    >
      Please select the file for Preview
    </div>
  </div>
</div>
`;

exports[`DeactivateAccount.js test snapshot test 1`] = `
<div>
  <h2>
    activate/deactivate account
  </h2>
  <button
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;

exports[`Home.js test snapshot test 1`] = `
<div>
  <h2>
    HELLO
  </h2>
  <img
    alt="fireSpot"
    height="200px"
    src="fb_default.jpg"
    width="200px"
  />
</div>
`;

exports[`Incoming.js test snapshot test 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Accept
  </button>,
  <button
    onClick={[Function]}
  >
    Reject
  </button>,
]
`;

exports[`Login.js test snapshot test 1`] = `<Login />`;

exports[`Main.js test snapshot test 1`] = `<Main />`;

exports[`MainView.js test snapshot test 1`] = `<MainView />`;

exports[`OnCall.js test snapshot test 1`] = `
<div
  className="call"
>
  <div
    className="hang-up"
  >
    <button
      color="red"
    >
      Hang up
    </button>
  </div>
</div>
`;

exports[`Phone.js test snapshot test 1`] = `
Array [
  <div
    className="call"
  >
    <button
      color="green"
      onClick={[Function]}
    >
      Call Test
    </button>
  </div>,
  <p
    className="status"
  >
    Connecting
  </p>,
]
`;

exports[`PostStatus.js test snapshot test 1`] = `<UNDEFINED />`;

exports[`UserSignUp.js test snapshot test 1`] = `<UserSignUp />`;

exports[`twilioStates.js test snapshot test 1`] = `<states />`;
